presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetColumnChunk.java

```diff
@@ -186,4 +230,16 @@ private boolean hasMorePages(long valuesCount, int pagesCount)
         return offsetIndex == null ? valuesCount < descriptor.getColumnChunkMetaData().getValueCount()
                 : pagesCount < offsetIndex.getPageCount();
     }
+
+    private boolean hasDictionaryPage(ColumnChunkMetaData columnChunkMetaData)
+    {
+        EncodingStats stats = columnChunkMetaData.getEncodingStats();
+        if (stats != null) {
+            return stats.hasDictionaryPages() && stats.hasDictionaryEncodedPages();
+        }
+        else {
```

Comment:
else not needed?

Reply:
In case of stats == null, we will rely on encodings

Reply:
yep, the logic is needed, shall we remove:
`else {`
since the if statement already returns